File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        nums = set(nums)
        
        longest_streak = 0
        
        for num in nums:
            if (num-1) not in nums:
                current_num = num
                current_streak = 1
        
                while (current_num + 1) in nums:
                    current_streak += 1
                    current_num = current_num + 1
                
                longest_streak = max(longest_streak, current_streak)
                
        return longest_streak

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var longestConsecutive = function(nums) {
    const map = {};
    let longest = 0;
    
    for (const num of nums) {
        map[num] = 1;
    }
    
    for (const num of nums) {
        if ((num-1) in map) continue;
        
        let currentLength = 1;
        let currentNum = num;
        
        while (currentNum+1 in map) {
            currentLength += 1;
            currentNum += 1;
        }
        
        longest = Math.max(longest, currentLength);
    }
    
    return longest;
};